generate_outputs.py

```diff
@@ -122,30 +116,26 @@
             log.debug(f"Loaded {len(messages)} messages")
             recovery_datasets.append(messages)
 
-    log.info("Loading demographics datasets:")
-    demogs_datasets = []
-    for i, demog_flow_name in enumerate(pipeline_configuration.demog_flow_names):
-        raw_survey_path = f"{raw_data_dir}/{demog_flow_name}.json"
+    log.info("Loading demographics + follow up surveys:")
+    survey_datasets = []
+    for i, survey_flow_name in enumerate(pipeline_configuration.demog_flow_names
+                                         + pipeline_configuration.follow_up_flow_names):
+        raw_survey_path = f"{raw_data_dir}/{survey_flow_name}.json"
         log.info(f"Loading {raw_survey_path}...")
         with open(raw_survey_path, "r") as f:
             contacts = TracedDataJsonIO.import_json_to_traced_data_iterable(f)
         log.debug(f"Loaded {len(contacts)} contacts")
-        demogs_datasets.append(contacts)
-
-    # Load Follow up Surveys
-    follow_up_survey_datasets = []
-    for i, follow_up_name in enumerate(pipeline_configuration.follow_up_flow_names):
-        raw_activation_path = f"{raw_data_dir}/{follow_up_name}.json"
-        log.info(f"Loading {raw_activation_path}...")
-        with open(raw_activation_path, "r") as f:
-            messages = TracedDataJsonIO.import_json_to_traced_data_iterable(f)
-        log.debug(f"Loaded {len(messages)} messages")
-        follow_up_survey_datasets.append(messages)
+        survey_datasets.append(contacts)
 
     # Add survey data to the messages
     log.info("Combining Datasets...")
+    coalesced_surveys_datasets = []
+    for dataset in survey_datasets:
+        coalesced_surveys_datasets.append(CombineRawDatasets.coalesce_traced_runs_by_key(user, dataset, "avf_phone_id"))
+
     data = CombineRawDatasets.combine_raw_datasets(user, messages_datasets + recovery_datasets,
-                                                   follow_up_survey_datasets, demogs_datasets)
+                                                   coalesced_surveys_datasets)
```

Comment:
Is this the right the thing to do be doing here? Unless I'm misunderstanding, it looks like you're switching from treating follow up flows as new rows in the dataset to columns

Reply:
Yes planning to switch to columns. Will treat follow ups as demogs in the next pr. Due to the number of radio shows treating follow ups as rows increases the already long headers.